### ISSUE TITLE
[FEATURE] Submit a testimonial from UI

### ISSUE THREAD

**Issue by amandamartin-dev** [EddieHubCommunity/LinkFree#2443]
### Description

Idea

From a linkfree profile testimonials tab, there should be a way for a profile visitor to click to add a testimonial.  This could simply open a new testimonial issue template for now

Future....
In the future, it could allow for submission of non-Github user testimonials

Concerns to consider (if we pursue a non GH version):
Approval flow for a testimonial
Authentication of a person providing the testimonial - for example, if they are not a GH user could they authenticate via linkedIn or twitter?

**Comment by eddiejaoude** [comment]
Great idea Amanda 👍  Let's do it 🚀  (in stages though lol)

**Comment by amandamartin-dev** [comment]
@eddiejaoude Oh my goodness yes! 

Okay, happy to take this on for phase 1 if you assign me - no rush won't start this evening (a button or link to submit a testimonial that opens up the GH issue template)

For the long term idea should I move it into it's own issue or is there a better place for future ideas not to be implemented just yet? (Allowing non GH users to submit)

**Comment by eddiejaoude** [comment]
I just realised there is an issue template already for people to submit testimonials for others https://github.com/EddieHubCommunity/LinkFree/issues/new?assignees=&labels=testimonial&template=testimonial.yml&title=New+Testimonial

Maybe we need to add this to the app so that people know it is there?

Yes for the future we will allow login and people to manage their account etc from the traditional form 👍

**Comment by amandamartin-dev** [comment]
Yes this is perfect! So I would want to use query params to populate that testimonial form with the user name from the profile they click "submit a testimonial" from so that the form is prepopulated as needed for the issue and is better UX

Since not all folks have testimonials yet, this link would have to be added in a universal location. 

Some ideas:
In the header or footer
Auto-added to the end of everyones link list

Example issue template
![testimonial with highlighted text to show adding query param to fill name in form](https://user-images.githubusercontent.com/97615019/209147385-a34cd566-783f-48d1-9fd1-7c6e3e2b53d5.PNG)

Example universal location for testimonial link - would be at the end of everyones link list automatically
![add testimonial idea showing a link at the end of folks linkefree link lists](https://user-images.githubusercontent.com/97615019/209147613-f933383d-eb8d-41fd-a9f0-a4f8fc077ec0.PNG)

**Comment by eddiejaoude** [comment]
> Example universal location for testimonial link - would be at the end of everyones link list automatically

OMG I love the idea of appending it to the links list 😍 

> from so that the form is prepopulated

That is a great idea also, especially for people's usernames! Is this possible with GitHub Issues?

**Comment by eddiejaoude** [comment]
What about... instead of adding another link for the `testimonials` link, it could be in the corner, similar to the `help` on the home page? 

<img width="493" alt="Screenshot 2022-12-22 at 18 29 47" src="https://user-images.githubusercontent.com/624760/209202846-6f8207ce-5806-437d-a75f-b107aaede823.png">

**Comment by amandamartin-dev** [comment]
Oh yes! Even better! I was in the profiles so didn't even consider that feature.  It will be more universal and not create any confusion in the links area.  Let's do it!

**Comment by amandamartin-dev** [comment]
re: query param to the issue tempalte. I thiiiink so.  I briefly googled before suggesting and saw some promising converastions but will need to properly investigate.

**Comment by eddiejaoude** [comment]
I could only find that it is possible but seems limited to the official fields, not the custom form fields https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue#creating-an-issue-from-a-url-query

**Comment by amandamartin-dev** [comment]
Hmmm, okay. May need to think on this for a minute bc from linkfree it won't necessarily be obvious what a users GitHub profile name is for the issue template. I'll look around later

**Comment by amandamartin-dev** [comment]
Good news! Although the docs don't say it will work, I just tested and IT DOES!

Check it out: https://github.com/EddieHubCommunity/LinkFree/issues/new?assignees=&labels=testimonial&template=testimonial.yml&title=New+Testimonial&name=Amanda+Test

![query-param-test](https://user-images.githubusercontent.com/97615019/209239589-912ef53a-f8d2-4a2a-b913-209c4679d9ed.PNG)

**Comment by amandamartin-dev** [comment]
@eddiejaoude check out the last comment and let me know if you want me to work on this implementation.  I submitted to github that this ability is missing from the docs, haven't heard back yet if there are limitations so lmk if you think we should wait to hear fully

**Comment by eddiejaoude** [comment]
Sorry I missed your previous comment with all my notifications 🤦‍♂️ - great find 👍 yes go for it 💯

**Comment by eddiejaoude** [comment]
How it looks

<img width="1049" alt="Screenshot 2023-01-03 at 18 24 51" src="https://user-images.githubusercontent.com/624760/210418401-3337352a-ee07-4b38-b929-f62209e27be4.png">

**Comment by amandamartin-dev** [comment]
Nice!
